CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -178,16 +193,20 @@ class SiStripGainFromCalibTree : public ConditionDBWriter<SiStripApvGain> {
 	std::string  OutputGains;
 	vector<string> VInputFiles;
 
-	MonitorElement*        Charge_Vs_Index;
-	MonitorElement*        Charge_Vs_Index_Absolute;
-	MonitorElement*        Charge_Vs_PathlengthTIB;
-	MonitorElement*        Charge_Vs_PathlengthTOB;
-	MonitorElement*        Charge_Vs_PathlengthTIDP;
-	MonitorElement*        Charge_Vs_PathlengthTIDM;
-	MonitorElement*        Charge_Vs_PathlengthTECP1;
-	MonitorElement*        Charge_Vs_PathlengthTECP2;
-	MonitorElement*        Charge_Vs_PathlengthTECM1;
-	MonitorElement*        Charge_Vs_PathlengthTECM2;
+        enum statistic_type {None=-1, StdBunch, StdBunch0T, FaABunch, FaABunch0T, IsoBunch, IsoBunch0T, Harvest};
+
+        std::vector<string>    dqm_tag_;
```

Comment:
for arrays - its memory overhead (obviously minor) and  bounds checking for free. Reserve is certainly needed for your vectors otherwise.